data/store/mongo/mongo_bucket_store_client.go

```diff
@@ -76,10 +76,19 @@ func (c *MongoBucketStoreClient) UpsertMany(ctx context.Context, userId *string,
 	// no data validation is done here as it is done in above layer in the Validate function
 	for _, sample := range samples {
 		ts := sample.GetTimestamp().Format("2006-01-02")
-		ops, _ := buildUpdateOneModel(dataType, sample, userId, ts, creationTimestamp)
-		operations = append(operations, ops...)
-
+		switch dataType {
+		case "Cbg":
+			ops, _ := buildCbgUpdateOneModel(sample, userId, ts, creationTimestamp)
+			operations = append(operations, ops...)
+		case "Basal":
+			ops, _ := buildBasalUpdateOneModel(sample, userId, ts, creationTimestamp)
+			operations = append(operations, ops...)
+		case "Bolus":
+			ops, _ := buildBolusUpdateOneModel(sample, userId, ts, creationTimestamp)
+			operations = append(operations, ops...)
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏`operations = append(operations, ops...)` could be moved outside the switch case to avoid repetition?

Reply:
it could be, it is done like this because it requires to declare the var "ops" outside of the loop. was afraid of any side effect